### PR TITLE
*: fix add index metrics

### LINKED ISSUE #31115: Improve some metrics on Grafana
## Enhancement

- [ ] TiDB client traffic is too complex, make it simply.
- [ ] `DESC` and `EXPLAIN` are used to show columns of a table and explain a SQL which should be separated on metrics.
- [ ] Add index duration and rate are merged into one panel, which makes the user confused.
- [ ] Batch client requests are displayed as a hate map, which is difficult to decide how many to change `max-batch-size`.

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve?

Issue Number: close #31115

Problem Summary:
Adding index metrics don't show the rate of adding index only, it combines with worker duration.

### What is changed and how it works?

1. Fix adding index metrics, maybe need to add worker duration in another PR.
2. Change 'speed' to 'rate' because 'speed' is a vector and 'rate' is a scalar.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
1. start a cluster
2. check the metrics
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by zimulala — COMMENTED** [review]
We need to modify `update_col_speed` too.

**Comment by jackysp** [comment]
> We need to modify `update_col_speed` too.

Done. I don't know why the file was not saved by vscode when I committed it.

**Comment by jackysp** [comment]
PTAL @wjhuang2016 @bb7133

File: metrics/grafana/tidb.json [hunk 1]
```diff
@@ -9263,26 +9263,19 @@
           "stack": false,
           "steppedLine": false,
           "targets": [
-            {
-              "expr": "histogram_quantile(1, sum(rate(tidb_ddl_batch_add_idx_duration_seconds_bucket{tidb_cluster=\"$tidb_cluster\"}[1m])) by (le, type))",
```
Comment (zimulala) [inline metrics/grafana/tidb.json:9267]: Why remove this？

Comment (jackysp) [reply metrics/grafana/tidb.json:9267]: This PR only fix adding index rate metrics, maybe need to add worker duration in another PR. The rate and duration should in two separate Grafana panels.

**Review by wjhuang2016 — APPROVED** [review]

**Comment by jackysp** [comment]
PTAL @zimulala

**Comment by jackysp** [comment]
PTAL @zimulala , thanks!

File: metrics/grafana/tidb.json (hunk 1 above)
Comment (zimulala) [reply metrics/grafana/tidb.json:9267]: So do we need to remove `tidb_ddl_batch_add_idx_duration_seconds_bucket` in `infoschema/metric_table_def.go`?

Comment (jackysp) [reply metrics/grafana/tidb.json:9267]: I don't think so. This metrics should be added back in the latter PR. However, the metrics adjustment requires changing the json file, and this kind of modification by adding a panel is not very controllable. So, I prefer to make a small change in this PR and add the panel in the next PR.

Comment (zimulala) [reply metrics/grafana/tidb.json:9267]: Got it.

**Review by zimulala — APPROVED** [review]
/LGTM

**Pull request merged**